src/appleseed.studio/mainwindow/pythonconsole/pythoneditor.cpp

```diff
@@ -78,6 +78,8 @@ void PythonEditor::keyPressEvent(QKeyEvent* event)
         insert_spaces(4);
     else
     {
+        if (event->key() == Qt::Key_Return || event->key() == Qt::Key_Enter)
```

Comment:
Did you find the root cause of the problem causing Shift+Enter inserting "lines within the line"?